pkg/api/types.go

```diff
@@ -2186,63 +2186,78 @@ type ProvisionScriptParametersInput struct {
+		"CNI_PLUGINS_URL":                      cloudSpecConfig.KubernetesSpecConfig.CNIPluginsDownloadURL,
+		"CLOUDPROVIDER_BACKOFF":                strconv.FormatBool(to.Bool(kubernetesConfig.CloudProviderBackoff)),
+		"CLOUDPROVIDER_BACKOFF_MODE":           kubernetesConfig.CloudProviderBackoffMode,
+		"CLOUDPROVIDER_BACKOFF_RETRIES":        strconv.Itoa(kubernetesConfig.CloudProviderBackoffRetries),
+		"CLOUDPROVIDER_BACKOFF_EXPONENT":       strconv.FormatFloat(kubernetesConfig.CloudProviderBackoffExponent, 'f', -1, 64),
+		"CLOUDPROVIDER_BACKOFF_DURATION":       strconv.Itoa(kubernetesConfig.CloudProviderBackoffDuration),
+		"CLOUDPROVIDER_BACKOFF_JITTER":         strconv.FormatFloat(kubernetesConfig.CloudProviderBackoffJitter, 'f', -1, 64),
+		"CLOUDPROVIDER_RATELIMIT":              strconv.FormatBool(to.Bool(kubernetesConfig.CloudProviderRateLimit)),
+		"CLOUDPROVIDER_RATELIMIT_QPS":          strconv.FormatFloat(kubernetesConfig.CloudProviderRateLimitQPS, 'f', -1, 64),
+		"CLOUDPROVIDER_RATELIMIT_QPS_WRITE":    strconv.FormatFloat(kubernetesConfig.CloudProviderRateLimitQPSWrite, 'f', -1, 64),
+		"CLOUDPROVIDER_RATELIMIT_BUCKET":       strconv.Itoa(kubernetesConfig.CloudProviderRateLimitBucket),
+		"CLOUDPROVIDER_RATELIMIT_BUCKET_WRITE": strconv.Itoa(kubernetesConfig.CloudProviderRateLimitBucketWrite),
+		"LOAD_BALANCER_DISABLE_OUTBOUND_SNAT":  strconv.FormatBool(to.Bool(kubernetesConfig.CloudProviderDisableOutboundSNAT)),
+		"USE_MANAGED_IDENTITY_EXTENSION":       strconv.FormatBool(kubernetesConfig.UseManagedIdentity),
+		"USE_INSTANCE_METADATA":                strconv.FormatBool(to.Bool(kubernetesConfig.UseInstanceMetadata)),
+		"LOAD_BALANCER_SKU":                    kubernetesConfig.LoadBalancerSku,
+		"EXCLUDE_MASTER_FROM_STANDARD_LB":      strconv.FormatBool(to.Bool(kubernetesConfig.ExcludeMasterFromStandardLB)),
+		"MAXIMUM_LOADBALANCER_RULE_COUNT":      strconv.Itoa(kubernetesConfig.MaximumLoadBalancerRuleCount),
+		"CONTAINER_RUNTIME":                    kubernetesConfig.ContainerRuntime,
+		"CONTAINERD_DOWNLOAD_URL_BASE":         cloudSpecConfig.KubernetesSpecConfig.ContainerdDownloadURLBase,
+		"KMS_PROVIDER_VAULT_NAME":              input.ClusterKeyVaultName,
+		"IS_HOSTED_MASTER":                     strconv.FormatBool(cs.Properties.IsHostedMasterProfile()),
+		"IS_IPV6_DUALSTACK_FEATURE_ENABLED":    strconv.FormatBool(cs.Properties.FeatureFlags.IsFeatureEnabled("EnableIPv6DualStack")),
+		"AUTHENTICATION_METHOD":                cs.Properties.GetCustomCloudAuthenticationMethod(),
+		"IDENTITY_SYSTEM":                      cs.Properties.GetCustomCloudIdentitySystem(),
+		"NETWORK_API_VERSION":                  APIVersionNetwork,
+		"NETWORK_MODE":                         kubernetesConfig.NetworkMode,
+		"KUBE_BINARY_URL":                      kubernetesConfig.CustomKubeBinaryURL,
+	}
+
+	keys := make([]string, 0)
+	for k := range parameters {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
```

Comment:
ooh sorted

Reply:
yeah you can't range through a map in order in go so it was that or rewrite the unit tests... But I think having a sorted result is nice so that's bonus